Doppler.BillingUser/Controllers/WebhooksController.cs

```diff
@@ -0,0 +1,91 @@
+            _currencyRepository = currencyRepository;
+            _logger = logger;
+            _userRepository = userRepository;
+            _mercadoPagoService = mercadoPagoService;
+            _paymentStatusMapper = paymentStatusMapper;
+        }
+
+        [HttpPost("/accounts/{accountname}/integration/mercadopagonotification")]
+        public async Task<IActionResult> UpdateMercadoPagoPaymentStatusAsync([FromRoute] string accountname, [FromBody] MercadoPagoNotification notification)
+        {
+            if (notification.Action != PAYMENT_UPDATED)
+            {
+                return new OkObjectResult("Successful");
+            }
+
+            var user = await _userRepository.GetUserInformation(accountname);
+            if (user is null)
+            {
+                return new NotFoundObjectResult("Account not found");
+            }
+
+            var invoice = await _billingRepository.GetInvoice(user.IdUser, notification.Data.Id.ToString());
+            if (invoice is null)
+            {
+                _logger.LogError("Invoice with authorization number: {0} was not found.", notification.Data.Id);
+                return new NotFoundObjectResult("Invoice not found");
+            }
+
+            var payment = await _mercadoPagoService.GetPaymentById(notification.Data.Id, accountname);
+
+            var status = _paymentStatusMapper.MapToPaymentStatus(payment.Status);
+            if (status == PaymentStatusEnum.Pending && invoice.Status == PaymentStatusEnum.Approved)
+            {
+                return new OkObjectResult("Successful");
+            }
+
+            if (status == PaymentStatusEnum.DeclinedPaymentTransaction && invoice.Status != PaymentStatusEnum.DeclinedPaymentTransaction)
+            {
+                if (invoice.Status == PaymentStatusEnum.Approved)
+                {
+                    _logger.LogError($"The payment associated to the invoiceId {invoice.IdAccountingEntry} was rejected. Reason: {payment.StatusDetail}");
+                }
+
+                await _billingRepository.UpdateInvoiceStatus(invoice.IdAccountingEntry, status);
+                return new OkObjectResult("Successful");
+            }
+
+            if (status == PaymentStatusEnum.Approved && invoice.Status != PaymentStatusEnum.Approved)
+            {
+                var accountingEntryMapper = new AccountingEntryForMercadopagoMapper(_currencyRepository);
+                var encryptedCreditCard = await _userRepository.GetEncryptedCreditCard(accountname);
+                var paymentEntry = await accountingEntryMapper.MapToPaymentAccountingEntry(invoice, encryptedCreditCard);
+                await _billingRepository.UpdateInvoiceStatus(invoice.IdAccountingEntry, PaymentStatusEnum.Approved);
+                await _billingRepository.CreatePaymentEntryAsync(invoice.IdAccountingEntry, paymentEntry);
+            }
+            return new OkObjectResult("Successful");
+        }
+        // TODO: Send emails
```

Comment:
Adding a blank lines after a block of code makes the code much more readable.
```suggestion
            }
            
            return new OkObjectResult("Successful");
        }
        
        // TODO: Send emails
```